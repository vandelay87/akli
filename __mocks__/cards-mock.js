export const cardMock = [
  {
    id: '1',
    title: 'Lorem Ipsum',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  }
];

export const cardWithSubtitleMock = [
  {
    id: '1',
    title: 'Lorem Ipsum',
    subtitle: 'Duis aute irure',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  }
];

export const cardsWithImageMock = [
  {
    id: '1',
    title: 'Facebook',
    subtitle: 'email@facebook.com',
    image: {
      file: {
        url: 'https://via.placeholder.com/200x120',
      }
    },
    description: 'Connect with friends, family and other people you know. Share photos and videos, send messages and get updates.',
  },
  {
    id: '2',
    title: 'Twitter',
    subtitle: 'email@twitter.com',
    image: {
      file: {
        url: 'https://via.placeholder.com/200x120',
      }
    },
    description: 'See what is happening in the world right now. Join Twitter today. Sign up Log In.',
  },
  {
    id: '3',
    title: 'LinkedIn',
    subtitle: 'email@linkedin.com',
    image: {
      file: {
        url: 'https://via.placeholder.com/200x120',
      }
    },
    description: 'Manage your professional identity. Build and engage with your professional network. Access knowledge, insights and opportunities.',
  },
];