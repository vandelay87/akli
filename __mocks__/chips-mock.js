// eslint-disable-next-line import/no-extraneous-dependencies
import { action } from '@storybook/addon-actions'

export const chipsWithActionMock = [
  {
    id: '1',
    value: 'Small',
    action: action('small-click'),
  },
  {
    id: '2',
    value: 'Medium',
    action: action('medium-click'),
  },
  {
    id: '3',
    value: 'Large',
    action: action('large-click'),
  },
]

export const chipsWithLinkMock = [
  {
    id: '1',
    value: 'React.js',
    link: 'https://reactjs.org/',
  },
  {
    id: '2',
    value: 'Vue',
    link: 'https://vuejs.org/',
  },
  {
    id: '3',
    value: 'Angular',
    link: 'https://angular.io/',
  },
  {
    id: '4',
    value: 'Gatsby',
    link: 'https://www.gatsbyjs.org/',
  },
]
