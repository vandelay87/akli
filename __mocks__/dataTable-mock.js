export const header = {
  rowData: [
    {
      id: 'itemTitle',
      value: 'Item',
    },
    {
      id: 'itemDescription',
      value: 'Description',
    },
    {
      id: 'priceTitle',
      value: 'Current price',
    },
    {
      id: 'stockTitle',
      value: 'In stock',
    },
  ],
}

export const rows = [
  {
    id: 'row1',
    rowData: [
      {
        id: 'coffeeLabel',
        value: 'Coffee',
      },
      {
        id: 'coffeeDescription',
        value: 'A nice cup of coffee. Hot!',
      },
      {
        isNumeric: true,
        id: 'coffeePrice',
        value: '£2.60',
      },
      {
        id: 'coffeeStock',
        value: 'Yes',
      },
    ],
  },
  {
    id: 'row2',
    rowData: [
      {
        id: 'tvLabel',
        value: 'TV',
      },
      {
        id: 'tvDescription',
        value: '42inch, 4k, widescreen television. Second hand.',
      },
      {
        isNumeric: true,
        id: 'tvPrice',
        value: '£399',
      },
      {
        id: 'tvStock',
        value: 'No',
      },
    ],
  },
  {
    id: 'row3',
    rowData: [
      {
        id: 'guitarLabel',
        value: 'Electric guitar',
      },
      {
        id: 'guitarDescription',
        value:
          'Limited edition electric guitar. Poor condition. Strings need to be changed.',
      },
      {
        isNumeric: true,
        id: 'guitarPrice',
        value: '£260',
      },
      {
        id: 'guitarStock',
        value: 'Yes',
      },
    ],
  },
]
