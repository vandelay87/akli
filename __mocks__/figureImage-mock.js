export const basicImage = {
  file: {
    url: 'https://via.placeholder.com/400x225',
  },
  description: 'An image.'
};

export const imageWithSize = {
  file: {
    url: 'https://via.placeholder.com/400x225',
    details: {
      image: {
        width: 400,
        height: 225
      }
    }
  },
  description: 'An image.'
};

export const fluidImage = {
  file: {
    url: 'https://via.placeholder.com/400x225',
    details: {
      image: {
        width: 400,
        height: 225
      }
    }
  },
  fluid: {
    aspectRatio: 1.5,
    src: `test_image.jpg`,
    srcSet: `some srcSet`,
    sizes: `(max-width: 600px) 100vw, 600px`,
  },
  description: 'An image.'
};