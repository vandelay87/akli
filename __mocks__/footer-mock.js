export const footerMock = [
  {
    id: '1',
    title: 'Home',
    link: '/',
  },
  {
    id: '2',
    title: 'Google',
    link: 'https://www.google.com',
    external: true,
  },
]
