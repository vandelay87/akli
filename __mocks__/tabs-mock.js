export const tabsMock = [
  {
    id: '1',
    label: 'tab 1',
    content: {
      id: 'c1',
      json: {
        content: [
          {
            content: [
              {
                marks: [],
                value: 'Tab 1 content',
                nodeType: 'text',
              },
            ],
            nodeType: 'heading-2',
          },
        ],
      },
    },
  },
  {
    id: '2',
    label: 'tab 2',
    content: {
      id: 'c2',
      json: {
        content: [
          {
            content: [
              {
                marks: [],
                value: 'Tab 2 content',
                nodeType: 'text',
              },
            ],
            nodeType: 'heading-2',
          },
        ],
      },
    },
  },
  {
    id: '3',
    label: 'tab 3',
    content: {
      id: 'c3',
      json: {
        content: [
          {
            content: [
              {
                marks: [],
                value: 'Tab 3 content',
                nodeType: 'text',
              },
            ],
            nodeType: 'heading-2',
          },
        ],
      },
    },
  },
]
