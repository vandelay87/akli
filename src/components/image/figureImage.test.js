import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import FigureImage from './figureImage'
import {
  imageWithoutCaption,
  imageWithCaption,
  imageWithContent,
} from './figureImage.stories'
import { fluidImage } from '../../../__mocks__/figureImage-mock'

describe('figureImage.js', () => {
  it('renders component', () => {
    const imageTree = renderer.create(imageWithoutCaption()).toJSON()
    const imageCaptionTree = renderer.create(imageWithCaption()).toJSON()
    const imageContentTree = renderer.create(imageWithContent()).toJSON()
    const imageFluidTree = shallow(<FigureImage image={fluidImage} />)

    expect(imageTree).toMatchSnapshot()
    expect(imageCaptionTree).toMatchSnapshot()
    expect(imageContentTree).toMatchSnapshot()
    expect(imageFluidTree).toMatchSnapshot()
  })
})
