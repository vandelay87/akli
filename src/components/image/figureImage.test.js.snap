// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`figureImage.js renders component 1`] = `
<figure
  className="sc-bdVaJa gfmnKv"
>
  <img
    alt="An image."
    src="https://via.placeholder.com/400x225"
  />
  
</figure>
`;

exports[`figureImage.js renders component 2`] = `
<figure
  className="sc-bdVaJa cJmbvo"
>
  <img
    alt="An image."
    src="https://via.placeholder.com/400x225"
  />
  <figcaption
    className="sc-bwzfXH kmIjOW"
  >
    A witty anecdote concerning the crocodile in the image above.
  </figcaption>
</figure>
`;

exports[`figureImage.js renders component 3`] = `
Array [
  <figure
    className="sc-bdVaJa eoVPHM"
  >
    <img
      alt="An image."
      src="https://via.placeholder.com/400x225"
    />
    
  </figure>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>,
]
`;

exports[`figureImage.js renders component 4`] = `ShallowWrapper {}`;
